### ISSUE TITLE
Deleting analysis through REST API also deletes machines from the DB

### ISSUE THREAD

**Issue by mkmanwich** [cuckoobox/cuckoo#624]
Hi,
When I delete an analysis using the REST API:
curl http://localhost:8090/tasks/delete/1
It also deletes all the machines that have the same tag as the deleted analysis.
When doing a machines/list API request it had some missing machines.

I confirmed it by using cuckoo with MySQL and deleting an analysis.
After delete, the "machines" table in the DB doesn't contain anymore all the machines that have the same tags as the analysis. (before delete it had 9 machines, now it has 4).
So its a real issue, caused by a faulty delete sql command that is sent to the DB when calling the delete API.

It's worth mentioning that restarting cuckoo "solves" the problem since it parses the configuration file again and adds all machines back.

**Comment by mkmanwich** [comment]
I did some debugging.
It seems the code is perfectly fine, issue seems to be with DB Handler (sqlalchemy):
When deleting an analysis record, it also tries to delete the tag associated with it from the "tags" table.
It causes everything linked to this tag to be deleted as well, including all other tasks that are tagged with that tag and all machines.

It is a cascading effect caused by trying to delete the analysis tag, which shouldn't be deleted in the first place. Interesting to note that the foreign keys are not configured with CASCADE but with RESTRICT. 

It is a serious issue, since among the other obvious issues it also causes severe un-syncing of DB and file system.

In short - delete function is completely broken, and is unusable until this is fixed.

**Comment by KillerInstinct** [comment]
This is actually already be tracked via:

https://github.com/cuckoobox/cuckoo/pull/573

**Comment by jbremer** [comment]
As @KillerInstinct said this is a duplicate of #573. I'll be pushing the proposed patch upstream soon! :)

**Comment by jbremer** [comment]
I've just removed the whole cascading thing so these side-effects should now have been taken care of. Closing this issue as it worked for me - please re-open or create a new issue if you have any new or other issues.
Thanks for reporting this bug @mkmanwich!

**Comment by mkmanwich** [comment]
I can confirm the fix works.

Thanks a lot :)
